src/OrbitGl/App.cpp

```diff
@@ -926,6 +1036,18 @@ std::string OrbitApp::GetCaptureTime() const {
   return GetPrettyTime(absl::Microseconds(time));
 }
 
+std::string OrbitApp::GetCaptureTimeAt(uint64_t timestamp_ns) const {
+  const TimeGraph* time_graph = GetTimeGraph();
+  if (time_graph == nullptr) {
+    return GetPrettyTime(absl::ZeroDuration());
+  }
+  const uint64_t capture_min_timestamp_ns = time_graph->GetCaptureMin();
+  if (timestamp_ns < capture_min_timestamp_ns) {
+    return GetPrettyTime(absl::ZeroDuration());
```

Comment:
A negative time could be useful to differentiate from the case where time_graph is null and to give and idea about how far from the capture min we are.

Reply:
Hmm, I don't think we should have show negative times. It can happen here if `GetCaptureMin()` is still returning `std::numeric_limits<uint64_t>::max()`, but ideally we should always have non-negative times here (it otherwise indicates an error on our side).

Reply:
The suggestion was to get more info on when the event occurred for debugging, but maybe we should be stricter here and actually CHECK since receiving an event from before the capture is an error on our side. Perhaps we could display the raw timestamp when we receive such events?

Reply:
Regarding debugging and raw timestamps, this is not really for our debugging, but to actually display information to the user, so we should keep it clean. For example, raw timestamps are timestamps since boot on the instance and they don't say a lot if we just print them. We should use `LOG` and `ERROR` for these things.

There is a more general problem with how we track capture min (and max) timestamps. The problem here is that `TimeGraph` is doing that and it shouldn't. That's why we can't really `CHECK` here. That is a bit bigger and out of scope from this change. I've created http://b/191631291 and I'll set this aside for now, as it doesn't really affect the capture log here.